src/conn/conn_ckpt.c

```diff
@@ -240,3 +240,38 @@ __wt_checkpoint_signal(WT_SESSION_IMPL *session, wt_off_t logsize)
         conn->ckpt_signalled = true;
     }
 }
+
+/*
+ * __wt_checkpoint_reserved_session_init --
+ *     Initialize checkpoint reserved session.
+ */
+int
+__wt_checkpoint_reserved_session_init(WT_SESSION_IMPL *session)
+{
+    WT_CONNECTION_IMPL *conn;
+
+    conn = S2C(session);
+
```

Comment:
I know we should only call this from one location but can we assert here that conn->ckpt_reserved_session is NULL?